tests/functional/watcher_controller_test.go

```diff
@@ -371,4 +387,32 @@ var _ = Describe("Watcher controller", func() {
 		})
 	})
 
+	When("Watcher is created  with no container images defined and env variables contains fake value", func() {
```

Comment:
It says "no container images defined and env variables contains fake value" but actually, you are using MinimalWatcherContainerSpec which defines images in the Spec parameters. So we have two options:

1. If we want to check that the resulting Watcher get the values from env variables when it's CRs do not includes images you need to CreateWatcher using MinimalWatcherSpec instead of MinimalWatcherContainerSpec.
2. If we want to check that the resulting Watcher get the values from CR even if env variables are set (which is what current test would be doing, then i think this should be "Watcher is created  with container images defined and env variables contains fake value". In this case the check should be:

It("should have the Spec fields with container images as defined in the CR")

As spec values wins to env variables.

My proposal would be to include these two checks separatedly.

Reply:
I failed to achieve the second case so skipping it for now.